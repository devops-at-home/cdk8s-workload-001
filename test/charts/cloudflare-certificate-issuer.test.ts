import { App } from 'cdk8s';
import { CloudflareCertificateIssuer } from '../../src/charts/cloudflare-certificate-issuer.chart';

const chart = 'cloudflare-certificate-issuer';

describe(chart, () => {
    test('Snapshot test', () => {
        const app = new App();

        const manifest = new CloudflareCertificateIssuer(app, chart, {
            namespace: 'cert-manager',
            labels: {
                chart,
            },
            environment: 'staging',
            cloudflare: {
                apiTokenSecretRef: {
                    name: 'apiTokenSecretRefName',
                    key: 'apiTokenSecretRefKey',
                },
                email: 'email',
            },
        });

        expect(manifest.toJson()).toMatchSnapshot();
    });
});
