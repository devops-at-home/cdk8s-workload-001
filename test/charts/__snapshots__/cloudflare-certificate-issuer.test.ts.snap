// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cloudflare-certificate-issuer Snapshot test 1`] = `
[
  {
    "apiVersion": "cert-manager.io/v1",
    "kind": "ClusterIssuer",
    "metadata": {
      "labels": {
        "chart": "cloudflare-certificate-issuer",
      },
      "name": "cloudflare-certificate-issuer-cloudflare-staging-c8168836",
      "namespace": "cert-manager",
    },
    "spec": {
      "acme": {
        "privateKeySecretRef": {
          "name": "letsencrypt-staging",
        },
        "server": "https://acme-staging-v02.api.letsencrypt.org/directory",
        "solvers": [
          {
            "dns01": {
              "cloudflare": {
                "apiTokenSecretRef": {
                  "key": "apiTokenSecretRefKey",
                  "name": "apiTokenSecretRefName",
                },
                "email": "email",
              },
            },
          },
        ],
      },
    },
  },
]
`;
